ophyd/runengine/runengine.py

```diff
@@ -158,22 +195,51 @@ def _start_scan(self, **kwargs):
                 if isinstance(det, SignalGroup):
                     # If we have a signal group, loop over all names
                     # and signals
+                    # print('vars(det) in ophyd _start_scan', vars(det))
                     for sig in det.signals:
-                        detvals.update({sig.name: sig.value})
+                        detvals.update({sig.name: {
+                            'timestamp': sig.timestamp[sig.pvname.index(sig.report['pv'])],
+                            'value': sig.value}})
                 else:
-                    detvals.update({det.name: det.value})
+                    detvals.update({
+                        det.name: {'timestamp': det.timestamp,
+                                   'value': det.value}})
             detvals.update(posvals)
+            detvals = mds.format_events(detvals)
             # TODO: timestamp this datapoint?
             # data.update({'timestamp': time.time()})
             # pass data onto Demuxer for distribution
-            print('datapoint[{}]: {}'.format(seqno, detvals))
-            event = data_collection.format_event(hdr, evdesc,
-                                                 seq_no=seqno,
-                                                 data=detvals)
-            create_event(event)
-            seqno += 1
+            print('datapoint[{}]: {}'.format(seq_num, detvals))
+            # grab the current time as a timestamp that describes when the
+            # event data was bundled together
+            bundle_time = time.time()
+            # actually insert the event into metadataStore
+            try:
```

Comment:
Wouldn't this be simpler to be

```python
if event_descritpor is None:
    event_descritpor = mds.whatever()

event = mds.insert_event(..)
```
so that the insertion code only is in there once?